build/allow_trust_test.go

```diff
@@ -0,0 +1,112 @@
+package build
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stellar/go-stellar-base"
+	"github.com/stellar/go-stellar-base/xdr"
+)
+
+var _ = Describe("Payment Mutators", func() {
+
+	var (
+		subject AllowTrustBuilder
+		mut     interface{}
+
+		address = "GAXEMCEXBERNSRXOEKD4JAIKVECIXQCENHEBRVSPX2TTYZPMNEDSQCNQ"
+		bad     = "foo"
+	)
+
+	JustBeforeEach(func() {
+		subject = AllowTrustBuilder{}
+		subject.Mutate(mut)
+	})
+
+	Describe("Trustor", func() {
+		Context("using a valid stellar address", func() {
+			BeforeEach(func() { mut = Trustor{address} })
+
+			It("succeeds", func() {
+				Expect(subject.Err).NotTo(HaveOccurred())
+			})
+
+			It("sets the destination to the correct xdr.AccountId", func() {
+				aid, _ := stellarbase.AddressToAccountId(address)
+				Expect(subject.AT.Trustor.MustEd25519()).To(Equal(aid.MustEd25519()))
+			})
+		})
+
+		Context("using an invalid value", func() {
+			BeforeEach(func() { mut = Trustor{bad} })
+			It("failed", func() { Expect(subject.Err).To(HaveOccurred()) })
+		})
+	})
+
+	Describe("SourceAccount", func() {
+		Context("using a valid stellar address", func() {
+			BeforeEach(func() { mut = SourceAccount{address} })
+
+			It("succeeds", func() {
+				Expect(subject.Err).NotTo(HaveOccurred())
+			})
+
+			It("sets the destination to the correct xdr.AccountId", func() {
+				aid, _ := stellarbase.AddressToAccountId(address)
+				Expect(subject.O.SourceAccount.MustEd25519()).To(Equal(aid.MustEd25519()))
+			})
+		})
+
+		Context("using an invalid value", func() {
+			BeforeEach(func() { mut = SourceAccount{bad} })
+			It("failed", func() { Expect(subject.Err).To(HaveOccurred()) })
+		})
+	})
+
+	Describe("AllowTrustAsset", func() {
+		Context("AssetTypeCreditAlphanum4", func() {
+			BeforeEach(func() {
+				mut = AllowTrustAsset{"USD"}
+			})
+
+			It("sets Asset properly", func() {
+				Expect(subject.AT.Asset.Type).To(Equal(xdr.AssetTypeAssetTypeCreditAlphanum4))
+				Expect(*subject.AT.Asset.AssetCode4).To(Equal([4]byte{'U', 'S', 'D', 0}))
+				Expect(subject.AT.Asset.AssetCode12).To(BeNil())
+			})
+		})
+
+		Context("AssetTypeCreditAlphanum12", func() {
+			BeforeEach(func() {
+				mut = AllowTrustAsset{"ABCDEF"}
+			})
+
+			It("sets Asset properly", func() {
+				Expect(subject.AT.Asset.Type).To(Equal(xdr.AssetTypeAssetTypeCreditAlphanum12))
+				Expect(subject.AT.Asset.AssetCode4).To(BeNil())
+				Expect(*subject.AT.Asset.AssetCode12).To(Equal([12]byte{'A', 'B', 'C', 'D', 'E', 'F', 0, 0, 0, 0, 0, 0}))
+			})
+		})
+	})
+
+	Describe("Authorize", func() {
+		Context("when equal true", func() {
+			BeforeEach(func() {
+				mut = Authorize{true}
+			})
+
+			It("sets authorize flag properly", func() {
+				Expect(subject.AT.Authorize).To(Equal(true))
+			})
+		})
+
+		Context("when equal false", func() {
```

Comment:
You might think about setting `subject.AT.Authorize` to `true` in your setup function, so you are confirming that the change is occurring.  The current form will always succeed because the zero value for `subject.AT.Authorize` is `false`.